src/instructlab/process/process.py

```diff
@@ -453,3 +454,80 @@ def get_latest_process() -> str | None:
     if not processes:
         return None
     return list(processes)[-1]
+
+
+def remove_process(process_uuid: str):
+    """Remove a process by its UUID and delete its log artifacts."""
+    process_registry = ProcessRegistry().load()
+    if process_uuid not in process_registry.processes:
+        logger.info(f"Process with UUID {process_uuid} not found.")
+        return
+
+    process = process_registry.processes.get(process_uuid)
+    log_file = process.log_path if process else None
+    stop_process(process_uuid, remove=True)
+    logger.debug(f"Stopped process {process_uuid}.")
+
+    # Remove the log
+    if log_file and os.path.exists(log_file):
+        os.remove(log_file)
+        logger.debug(f"Log file {log_file} removed.")
+
+    logger.info(f"Process with UUID {process_uuid} removed.")
+
+
+def filter_process_with_conditions(
+    process_uuid: Optional[str] = None,
+    state: Optional[str] = None,
+    older: Optional[int] = None,
+) -> List[str]:
+    """Remove processes based on their uuid, state or older."""
+    process_registry = ProcessRegistry().load()
+
+    process_records_to_remove = []
+    for item_uuid, process in process_registry.processes.items():
+        if process_uuid and item_uuid != process_uuid:
+            continue
+
+        if state and process.status != state:
+            continue
+
+        if older == 0:
+            process_records_to_remove.append(item_uuid)
+            continue
+
+        if older:
+            start_time = process._start_time
```

Comment:
If you need this attribute **publicly**, then please expose it using a public property. (Or introduce a public function in the Process class to return `time since start`.)